src/core/CorePool.ts

```diff
@@ -311,11 +343,63 @@ export class CorePool {
     tickUpper: number,
     liquidityDelta: JSBI
   ): { position: Position; amount0: JSBI; amount1: JSBI } {
-    // TODO
+    let amount0: JSBI = ZERO,
+      amount1: JSBI = ZERO;
+
+    let position: Position = this.getPosition(owner, tickLower, tickUpper);
```

Comment:
Let's add a conditional expression here to return instantly if liquidity balance is not enough for burn request.
```
if(JSBI.lessThan(liquidityDelta, ZERO)){
    const negatedLiquidityDelta = JSBI.multiply(liquidityDelta, NEGATIVE_ONE);
    assert(JSBI.greaterThanOrEqual(position.liquidity, negatedLiquidityDelta), "Liquidity Underflow");
}
```

Reply:
Nice solution